### DISCUSSION TITLE
Help with Simpson's Diversity Project 1

### DISCUSSION THREAD

**Discussion by hollycormack21** [IBAHCM/RPiR#483] [Archive] [answered]
I'm attempting to do project 1. I'm finding it a little difficult to write the Simpson's Diversity function. 
I initially thought to simply write out the equation: 
 D = sum(n/N)^2 
Using tree.pop and num.trees as n and N. However, this didn't allow functions to be passed through it. 
I now think that I have to set up n and N as arguments within the functions and define them within it. 
However, I'm not really sure how to write out an equation to calculate the number of individuals within the species and the total population. 
I imagine it will use a for loop, I'm not sure which arguments to use though I assume I shall still be using the vectors previously set up (tree.pop, num.trees, species.names, num.sp etc). 
Any help would be appreciated. 
Thank you

**Comment by richardreeve** [comment, marked as answer]
@hollycormack21 - I'm sorry that you're obviously struggling a bit here. Your calculation of Simpson index doesn't quite do things in the right order - the exercise says "squares them all individually, and add them all together" but you add them all up and then square them, so you need to fix that. If you can't remember how to write a function in general, you can try practical A-2 again, and maybe A-5, but you have been writing functions throughout the course so I hope you can refer to some of your earlier work. These initial functions just need one argument, which is the population to study - you should be able to calculate the total size of the population from that (using `sum()` as above). You could also go back to the recording of the last Zoom meeting before Christmas to see more about how to do the first project exercise.
